pulsar/producer_partition.go

```diff
@@ -430,26 +432,74 @@ type pendingItem struct {
 	sync.Mutex
 	batchData    internal.Buffer
 	sequenceID   uint64
-	sentAt       int64
+	sentAt       time.Time
 	sendRequests []interface{}
 	completed    bool
 }
 
 func (p *partitionProducer) internalFlushCurrentBatch() {
+	if p.options.SendTimeout > 0 {
```

Comment:
How can this be disabled? Setting it to -1?

Reply:
Yes, negative to disable, such as -1 typically.
https://github.com/apache/pulsar-client-go/blob/master/pulsar/producer.go#L81